docs/source/user-guide/dataset_object.rst

```diff
@@ -3,76 +3,74 @@
+ - **label** - The classes of the classification problem or values of a regression problem.
+ - **index_name** - This can be useful to track indexes duplicate etc. Like the other arguments, this is not mandatory.
+ - **date_name** - The date the sample was acquired (Useful for TimeSeries problems). This column is standardized using `pd.Timestamp`.
+ - **features** - Specifies the columns used by model for training. Used for defining only a subset of the columns in the data as features. If not defined, all columns that aren't explicitly specified as `label`, `date`, or `index` are considered as features.
+
+All these and more can be defined and accessed on the Dataset Object.
+This allows the user the flexibility to define these column names whichever they seem fit,
+and allows use of the these tags to standardize development around this data.
+
+
+The Basics
+==========
+
+Lets start with building a simple pd.DataFrame that we might already know as a popular data structure to store data.
+But we also have other metadata that we have other use for, and not only as features for the model.
+
+    >>> d = {"id": [1,2,3,4],
+    ...      "feature1": [0.1,0.3,0.2,0.6],
+    ...      "feature2": [4,5,6,7],
+    ...      "cat_feature": [0,0,0,1],
+    ...      "class": [1,2,1,2]}
+    ... df = pd.DataFrame(d)
+    ... dataset = Dataset(df, label="class", index_name="id", cat_features=["category"])
+
+Now we have built a Dataset that we can easily pull features and labels from:
+
+    >>> dataset.features
+    ['feature1', 'feature2', 'category']
+    >>> dataset.label_name
+    ['class']
+
+We can also get pd.DataFrames from columns that interest us e.g.:
+
+    >>> dataset.features_columns
+        feature1	feature2	category
+    0	0.1	        4	        0
+    1	0.3	        5	        0
+    2	0.2	        6	        0
+    3	0.6	        7	        1
+
+Now we can see that only features columns are pulled, which is save us the hassle
+of tracking externally what is a feature column and what isn't one.
+
+We can see the same thing happening here with the label_column. Instead of tracking an X DataFrame and a y DataSeries,
+we have it all in one single object:
+
+    >>> dataset.label_col
+        class
+    0	1
+    1	2
+    2	1
+    3	2
+
+
+now instead of needing keeping track fo the label column, we can track it within the the Dataset Object.
 
 Inferring Features And Categorical Features
-================================================
+===========================================
 Dataset defines which columns of the data are features and of them which are categorical features.
 For features, if parameter `features` not passed explicitly, all will be considered features apart from the label, index and date.
 For categorical features, if parameter `cat_features` not passed explicitly, the following logic runs on every column to determine
```

Comment:
and replace sentence after (which it won't let me review on since it's not part of this PR, as:
how it will be treated.

(instead of `whether the column is categorical or not:`)